x-pack/legacy/plugins/uptime/common/runtime_types/ping/ping.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+
+export const HttpResponseBodyType = t.partial({
+  bytes: t.number,
+  content: t.string,
+  content_bytes: t.number,
+  hash: t.string,
+});
+
+export type HttpResponseBody = t.TypeOf<typeof HttpResponseBodyType>;
+
+export const TlsType = t.partial({
+  certificate_not_valid_after: t.string,
+  certificate_not_valid_before: t.string,
+});
+
+export type Tls = t.TypeOf<typeof TlsType>;
+
+export const PingType = t.intersection([
+  t.type({
+    '@timestamp': t.string,
+    monitor: t.intersection([
+      t.type({
+        duration: t.type({
+          us: t.number,
+        }),
+        id: t.string,
+        status: t.string,
+        type: t.string,
+      }),
+      t.partial({
+        check_group: t.string,
+        ip: t.string,
+        name: t.string,
+        timespan: t.partial({
+          gte: t.string,
+          lte: t.string,
+        }),
+      }),
+    ]),
+  }),
```

Comment:
Going forward we can declare monitor as a separate type so that it can be rused elsewhere.

Reply:
👍 I'll make the change in this PR.

Reply:
Done here https://github.com/elastic/kibana/pull/59392/commits/55ac2aab4dae9f98bdf4eeda8c99542b5826155a.